discretize/mixins/mpl_mod.py

```diff
@@ -0,0 +1,2070 @@
+            err += ' Mesh provided has {} dimension(s).'.format(mesh.dim)
+            raise ValueError(err)
+
+        # (b) v_type  # Not yet working for ['CCv']
+        v_typeOpts = ['CC', 'Fx', 'Fy', 'Fz', 'Ex', 'Ey', 'Ez']
+        if v_type not in v_typeOpts:
+            err = "v_type must be in ['{0!s}'].".format("', '".join(v_typeOpts))
+            err += " v_type provided: '{0!s}'.".format(v_type)
+            raise ValueError(err)
+
+        if v_type != 'CC':
+            aveOp = 'ave' + v_type + '2CC'
+            Av = getattr(mesh, aveOp)
+            if v.size == Av.shape[1]:
+                v = Av * v
+            else:
+                v = mesh.reshape(v, v_type[0], v_type) # get specific component
+                v = Av * v
+
+        # (c) vOpts  # Not yet working for 'vec'
+
+        # Backwards compatibility
+        if view in ['xy', 'yx']:
+            axis = view
+            view = 'real'
+
+        viewOpts = ['real', 'imag', 'abs']
+        if view in viewOpts:
+            v = getattr(np, view)(v) # e.g. np.real(v)
+        else:
+            err = "view must be in ['{0!s}'].".format("', '".join(viewOpts))
+            err += " view provided: '{0!s}'.".format(view)
+            raise ValueError(err)
+
+        # 1. Store relevant data
+
+        # Store data in self as (nx, ny, nz)
+        self.v = mesh.reshape(v.reshape((mesh.nC, -1), order='F'), 'CC', 'CC', 'M')
+        self.v = np.ma.masked_where(np.isnan(self.v), self.v)
+
+        # Store relevant information from mesh in self
+        self.x = mesh.grid_nodes_x    # x-node locations
+        self.y = mesh.grid_nodes_y    # y-node locations
+        self.z = mesh.grid_nodes_z    # z-node locations
+        self.xc = mesh.grid_cell_centers_x  # x-cell center locations
+        self.yc = mesh.grid_cell_centers_y  # y-cell center locations
+        self.zc = mesh.grid_cell_centers_z  # z-cell center locations
+
+        # Axis: Default ('xy'): horizontal axis is x, vertical axis is y.
+        # Reversed otherwise.
+        self.yx = axis == 'yx'
+
+        # Store initial slice indices; if not provided, takes the middle.
+        if xslice is not None:
+            self.xind = np.argmin(np.abs(self.xc - xslice))
+        else:
+            self.xind = self.xc.size // 2
+        if yslice is not None:
+            self.yind = np.argmin(np.abs(self.yc - yslice))
+        else:
+            self.yind = self.yc.size // 2
+        if zslice is not None:
+            self.zind = np.argmin(np.abs(self.zc - zslice))
+        else:
+            self.zind = self.zc.size // 2
+
+        # Aspect ratio
+        if isinstance(aspect, (list, tuple)):
+            aspect1 = aspect[0]
+            aspect2 = aspect[1]
+        else:
+            aspect1 = aspect
+            aspect2 = aspect
+        if aspect2 in ['auto', 'equal']:
+            aspect3 = aspect2
+        else:
+            aspect3 = 1.0/aspect2
+
+        # Get color limits.
+
```

Comment:
Feel free to copy paste my changes in #228 just in here and close that PR.

Reply:
yeah, with all of the interfaces now having a common call, it'll be a lot easier to make the change consistent.